pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -506,8 +506,8 @@ func (r *ReconcileClusterPool) reconcileRunningClusters(
 		cdList,
 		func(i, j int) bool {
 			return cdList[i].CreationTimestamp.Before(&cdList[j].CreationTimestamp) ||
-				cdList[i].Namespace < cdList[j].Namespace ||
-				cdList[i].Name < cdList[j].Name
+				cdList[i].CreationTimestamp.Equal(&cdList[j].CreationTimestamp) &&
+					cdList[i].Namespace < cdList[j].Namespace
```

Comment:
Thanks for fixing this, @fxierh! I agree it's busted. However, based on experiences I've had since this code was written, I would prefer to see an algorithm like:

```go
			if !cdList[i].CreationTimestamp.Equal(&cdList[j].CreationTimestamp) {
				return cdList[i].CreationTimestamp.Before(&cdList[j].CreationTimestamp)
			}
			return cdList[i].Namespace < cdList[j].Namespace
```

Personally I find this form easier to read/understand.

Reply:
Thanks for the review. The code is updated following the comment above.